src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/DefaultRazorProjectHost.cs

```diff
@@ -104,8 +97,16 @@ internal async Task OnProjectChanged(IProjectVersionedValue<IProjectSubscription
                     if (TryGetConfiguration(update.Value.CurrentState, out var configuration))
                     {
                         TryGetRootNamespace(update.Value.CurrentState, out var rootNamespace);
+
                         var hostProject = new HostProject(CommonServices.UnconfiguredProject.FullPath, configuration, rootNamespace);
 
+                        if (TryGetIntermediateOutputPath(update.Value.CurrentState, out var intermediatePath))
```

Comment:
I'd throw this entire piece in the `UpdateAsync` method in `RazorProjectHostBase` because that method seems to switch to the main thread before doing work. That way you wouldn't need to switch twice.